extensions/amp-gist/0.1/amp-gist.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Embeds a Github gist
+ *
+ * Example:
+ * <code>
+ * <amp-gist
+ *   layout="responsive"
+ *   data-gistid="a19e811dcd7df10c4da0931641538497"
+ *   width="100"
+ *   height="100">
+ * </amp-gist>
+ * </code>
+ */
+
+import {getIframe} from '../../../src/3p-frame';
+import {listenFor} from '../../../src/iframe-helper';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {removeElement} from '../../../src/dom';
+import {isExperimentOn} from '../../../src/experiments';
+import {user} from '../../../src/log';
+
+const TAG = 'amp-gist';
+
+export class AmpGist extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?HTMLIFrameElement} */
+    this.iframe_ = null;
+  }
+
+  /**
+  * @param {boolean=} opt_onLayout
+  * @override
+  */
+  preconnectCallback(opt_onLayout) {
+    this.preconnect.url('https://gist.github.com/', opt_onLayout);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    user().assert(isExperimentOn(this.win, TAG),
+        `Experiment "${TAG}" is disabled.`);
+  }
+
+  /** @override */
+  layoutCallback() {
+    /* the third parameter 'github' ties it to the 3p/github.js */
+    const iframe = getIframe(this.win, this.element, 'github');
+    this.applyFillContent(iframe);
+    // Triggered by window.context.requestResize() inside the iframe.
+    listenFor(iframe, 'embed-size', data => {
+      this./*REVIEW*/changeHeight(data.height);
```

Comment:
@levidurfee @jridgewell This needs to stay `changeHeight` similar to `amp-facebook` and `amp-instagram`, and a few other places. The reason is that this is a highly dynamic widget and if height does not change, the widget will be completely unusable, so change height is not optional in this case.